app/src/main/java/com/io/moviO/moviesList/MoviesListFragment.kt

```diff
@@ -20,7 +22,14 @@ class MoviesListFragment : Fragment(R.layout.fragment_movies_list),
 
         super.onViewCreated(view, savedInstanceState)
         viewModel.movies.observe(viewLifecycleOwner) {
-            apadter.updateMovieList(it)
+            when (it) {
+                is DataResult.Success -> apadter.updateMovieList(it.value)
+                is DataResult.Fail -> Toast.makeText(
+                    this.context,
+                    "Something goes wrong!",
```

Comment:
It would be better if this string is in your strings.xml and then you can reuse it on both screens.